src/request-data/resource.js

```diff
@@ -279,8 +279,20 @@ class Resource extends BaseResource {
 }
 
 const proxyHandler = {
-  get: (resource, prop) => {
-    if (prop in resource) return resource[prop];
+  get: (resource, prop, proxy) => {
+    // First check the resource for the property.
+    if (prop in resource) {
+      // We want to call withHttpMethods() on Resource.prototype.request().
+      // However, because the request() method references `this`, we must first
+      // bind it to the proxy. For example, if resource.request.post() is
+      // called, we need `this` to be the proxy in post().
+      if (prop === 'request')
+        return withHttpMethods(resource.request.bind(proxy));
```

Comment:
I wasn't sure at first where to do this binding. Initially, I put it in the constructor of the `Resource` class above. However, that binds the `request()` method to the `Resource` instance. Instead, we want to bind it to the resource proxy. The resource proxy is what is returned from this file: the file only uses the `Resource` instance internally.